### PR TITLE
Make oauth2 implementation more compliant

### PR DESCRIPTION
**Pull request by grimpy**
Pass state variable
Pass arguments as post data (form)
Pass grant_type empty
Do not pass client_secret in headers
Make it posible to configure redirect_uri

### CODE REVIEW

**Comment by khenderick** [comment]
I'm not sure how this change is in any way compliant with RFC 6749. The changed code is (supposed to be) part of https://tools.ietf.org/html/rfc6749#section-4.1 step D and E but would after this change be invalid.

**Comment by grimpy** [comment]
Fair remarks (altough some of those things nobody does), Ill make some adjustments in the course of this week.

**Comment by khenderick** [comment]
Where exactly do you see non-compliant code, I've double checked the code and I can't see anything that might be missing.

**Comment by grimpy** [comment]
params should (atleast in most common implementations) be passed via post data
request_uri instead of request_url (i instead of l)
the request_uri will moslty likely be some public domain not the ip of the storage node

**Comment by khenderick** [comment]
Makes sense indeed. But the custom configurable ```redirect_uri``` should also be specified in the root of the API since that's used by the first three steps of the flow.

**Comment by grimpy** [comment]
I've modified the changes. Now the only real change is that data is in postdata instead of query params.
And redirect_uri is written correct and passed a sane default.
redirect_uri is actually optional when requesting access_token (in our oauth2 server ignored anyway)

File: webapps/api/oauth2/redirectview.py [hunk 1]
```diff
@@ -67,14 +67,12 @@ def get(self, request, *args, **kwargs):
         base_url = Configuration.get('/ovs/framework/webapps|oauth2.token_uri')
         client_id = Configuration.get('/ovs/framework/webapps|oauth2.client_id')
         client_secret = Configuration.get('/ovs/framework/webapps|oauth2.client_secret')
+        redirect_uri = 'https://{0}/api/oauth2/redirect/'.format(request.META['HTTP_HOST'])
         parameters = {'grant_type': 'authorization_code',
-                      'redirect_url': 'https://{0}/api/oauth2/redirect/'.format(System.get_my_storagerouter().ip),
-                      'client_id': client_id,
```
Comment (khenderick) [inline webapps/api/oauth2/redirectview.py:72]: Shouldn't the ```client_id``` be part of the call cfr https://tools.ietf.org/html/rfc6749#section-4.1.3?

Comment (grimpy) [reply webapps/api/oauth2/redirectview.py:72]: REQUIRED, if the client is not authenticating with the
         authorization server as described in Section 3.2.1.

If you jump to seciont 3.2.1 and from there to 2.3 you will see its not required if you are passing the client_secret via authorization header.

Comment (khenderick) [reply webapps/api/oauth2/redirectview.py:72]: Aha, ok, thanks

**Review by khenderick — APPROVED** [review]

**Comment by khenderick** [comment]
Packaged in openvstorage-2.7.6-rev.4325.b08f4a0

**Pull request merged**